### ISSUE TITLE
The `match?` assert-expr (used in `is`) should return `true` or `false`

### ISSUE THREAD

**Issue by dchelimsky** [nubank/matcher-combinators#109]
``` clojure
(require '[matcher-combinators.standalone :as standalone]
         '[matcher-combinators.matchers :as matchers]
         '[matcher-combinators.clj-test]  ;; load `match?` assert-expr
         '[clojure.test :refer [is]])

(standalone/match? (matchers/equals 37) 37)
;; => true
(standalone/match? (matchers/equals 37) 42)
;; => false

(is (match? (matchers/equals 37) 37))
;; => nil
(is (match? (matchers/equals 37) 42))
;; => nil
```

This is confusing when evaluating inner expressions in order to debug.

**Comment by philomates** [comment]
oh, I didn't realize that `is` expressions returned values. This definitely makes sense to fix :+1:

**Comment by dchelimsky** [comment]
Fixed in #111
